tests/test_graph_components/providers/test_domain_without_responses_provider.py

```diff
@@ -0,0 +1,60 @@
+from typing import Text
+
+import pytest
+from rasa.graph_components.providers.domain_without_response_provider import (
+    DomainWithoutResponseProvider,
+)
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import KEY_E2E_ACTIONS, KEY_RESPONSES, Domain
+
+
+@pytest.mark.parametrize(
+    "domain_yml",
+    [
+        ("data/test_domains/travel_form.yml"),
+        ("data/test_domains/mixed_retrieval_intents.yml"),
+    ],
+)
+def test_copy_without_responses(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    domain_yml: Text,
+):
+
+    component = DomainWithoutResponseProvider.create(
+        {"arbitrary-unused": 234},
+        default_model_storage,
+        Resource("xy"),
+        default_execution_context,
+    )
+
+    domain = Domain.from_file(path=domain_yml,)
+    domain_copy = component.recreate_without_responses(domain=domain,)
+
+    given_as_dict = domain.as_dict()
+    copy_as_dict = domain_copy.as_dict()
+
+    # all configurations not impacted by responses stay intact
+    assert sorted(given_as_dict.keys()) == sorted(copy_as_dict.keys())
+    for key in given_as_dict.keys():
+        if key not in [KEY_RESPONSES, KEY_E2E_ACTIONS]:
+            assert given_as_dict[key] == copy_as_dict[key]
+
+    # Note: The given domains do contain responses and some actions for which no
+    # responses are defined...
+    assert domain.responses, "choose a different test config"
+    assert domain.user_actions, "choose a different test config"
+    assert set(domain.user_actions).difference(
+        domain.responses
+    ), "choose a different test config"
+
+    # Assert that the recreated copy does not contain any response information
+    assert not domain_copy.responses
+    assert all(action in domain._custom_actions for action in domain_copy.user_actions)
+    assert set(domain_copy.action_names_or_texts) < set(domain.action_names_or_texts)
+    assert all(
+        response not in domain_copy.action_names_or_texts
+        for response in set(domain.responses).difference(domain._custom_actions)
```

Comment:
I don't think you need the `difference` method here, since in theory `responses` and `_custom_actions` should be disjoint? Also I'd maybe add a similar assertion to check that `responses` are not in `domain_copy.user_actions`?

Reply:
I think, `_custom_actions` and `responses` could overlap - because to create `user_actions` we de-duplicate the concatenation of both. But maybe I'm overlooking some earlier check here?

Reply:
> Also I'd maybe add a similar assertion to check that responses are not in domain_copy.user_actions?
good point, will add that

Reply:
> _custom_actions and responses could overlap

According to [this](https://github.com/RasaHQ/rasa/blob/6b26632e7785544cc3d486c2628b832a8c474dc5/rasa/shared/core/domain.py#L217), only custom actions listed in the `actions` section of the domain fill the `action_names` argument  in the constructor which is what `self._custom_actions` gets assigned. But I don't know where else this constructor gets used.

Reply:
That is a nice and confusing (for me) observation 😅 . 

> But I don't know where else this constructor gets used.

Agree ^^. Then let's leave the ".difference" in there since the `_combine_with_responses` suggests that might happen somehow and the constructor would allow it - even if the methods loading those yamls would fix that (at the moment) for us?

Reply:
Yup, all good 👍🏼